src/NextGen/DonationForm/Blocks/DonationFormBlock/resources/app/templates/fields/Text.tsx

```diff
@@ -1,9 +1,9 @@
 import type {FieldProps} from '@givewp/forms/propTypes';
 
-export default function Text({label, placeholder, fieldError, inputProps, validationRules}: FieldProps) {
+export default function Text({requiredLabel, placeholder, fieldError, inputProps}: FieldProps) {
     return (
         <label>
-            <span>{label}</span>
+            <span>{requiredLabel}</span>
```

Comment:
This `requiredLabel` prop is confusing me.  What if a field is not required? wouldn't we need a conditional here to display the `label` string instead of the `requiredLabel` jsx?

Reply:
Yeah, I agree that it's a little weird. The problem I was trying to solve was introducing the accessible asterisk _without_ having to make every single template file handle that itself. I thought about simply making a `<RequiredLabel />` component that template components could use, but realized we'd need to make it easily accessible to 3rd parties, and it's adding even more mental complexity for something simple. I then thought to just update the label itself, so folks don't worry about it at all! But then I realized it's important to preserve the label as a simple string for things like error messages. So then I arrived at making a new label prop that template components would simply use in place of the label and just not worry about it.

Perhaps this could be flipped on its head. I could take the original label, assign it to a `Field.rawLabel` (or something like) property, and then update `Field.label`. Then the template components would just use `label` as they have. We'd then have `rawLabel` to use internally for things like messages.

Reply:
I hear you. I think it's useful to generate common html like this for templates I just want to keep the compositional templating in tact. Here's an example of what I would be happier to see inside a field.

`<span>{isRequired ? labelWithRequiredIndicator : label}</span>`

Reply:
I'm using this to think through this one last time, @jonwaldstein :

### Method 1: Replace the Label
- Update the `Field.label` so it includes the indicator when necessary.
- Add a `Field.plainLabel` which is always the simple, text version of the label (no indicator)

**Pros**
- Templates do nothing special to display the requirement indicator
- Requirement remains a validation concept and isn't used in the template

**Cons**
- Template has to know about the `Field.plainLabel` if they for some reason want to do something with the label and not show the indicator

### Method 2: Include a new labelWithRequiredIndicator property
- Add a `Field.isRequired` property as a shorthand for `Field.validationRules.required` _(optional)_
- Add a `Field.labelWithRequiredIndicator` property which templates can use

**Pros**
- Whether or not the plain label or label with indicator is used is clear in the template and up to the template
- The `Field.label` is predictably always just the plain label string

**Cons**
- Every template has to include the condition as to which form of the label to display based on requirement
- We risk template overrides not realizing they need to do this ❗
- Field templates must think about validation in order to render properly

---

Ok. That really helped! I'm really leaning towards Method 1 at this point. Running various scenarios through my head, I can't actually think of a reason (or at least common one) wherein someone would want to hide the indicator when rendering the field. If they do, then they're probably overriding, and frankly it would be simpler to just hide the indicators via CSS (which would keep them for accessibility). Method 2 ends up putting the onus on each template (and override) to include the same condition. As noted, I think this is more likely to create scenarios wherein folks get confused why the indicator isn't showing up — not realizing its up to them. It also begs the question as to whether `Field.labelWithRequiredIndicator` is null or without the indicator if the field isn't required, which opens up more room for issues/confusion.

Unless you can think of a serious counterpoint (which I'm open to) I'm going to move forward with Method 1. In any case, it's not what I started with, so thank you for thinking through this with me!

Reply:
Thanks for writing this down. 

So there's 1 main scenario that I think could warrant another look at method 2 or even another solution that would satisfy them.

**Scenario**
Template wants to change the asterisk to the word "required".  
   - example: `First name (required)` vs `First name *`

The asterisk and word 'required' are both recommended ways of providing accessible required indicators for a label.  I think it would be valuable for templates to have control over which they prefer to display instead of being locked into one or the other.

Also, I think this discussion should include field errors as well.  Notice how we are handling errors in fields by using a conditional for `fieldError` and then displaying some html.  Whatever direction we decide for labels should reflect other conditional template logic like field errors.

_As I was writing this I just came up with this idea..._

Here's one more method to throw in the mix:

**Method 3: extract common field props into their own template (derived from FieldNode)** 

With this method we could still have the simplicity of **method 1** where the template would just display {label} and {fieldError} but with the flexibility of overriding the html for all fields. 🤯 

```
function Label({label, required}) {
    return (
        <>
            {label}{' '}
            {required && (
                <span className="givewp-field-required" title="Field Required">
                    *
                </span>
            )}
        </>
    );
}

function FieldError({error}) {
    return (
        error && (
            <div className="error-message">
                <p role="alert">{error}</p>
            </div>
        )
    );
}

template = {
    fieldProps: {
        label: Label,
        fieldError: FieldError
    }
}
```

Reply:
We've done it. 😎